bitcoin/src/merkle_tree/block.rs

```diff
@@ -276,7 +276,7 @@ impl PartialMerkleTree {
             return Err(NoTransactions);
         };
         // check for excessively high numbers of transactions
-        if self.num_transactions > MAX_BLOCK_WEIGHT / MIN_TRANSACTION_WEIGHT {
+        if self.num_transactions as u64 > (Weight::MAX_BLOCK / Weight::MIN_TRANSACTION.to_wu()).to_wu() {
```

Comment:
`Div<Weight>` could have been useful here...

Reply:
Why it's not `Weight::MAX_BLOCK.to_wu() / Weight::MIN_TRANSACTION.to_wu()`?

Anyway, I wouldn't mind having `Div` since it properly returns unitless number but it might look a bit strange.

Reply:
> Why it's not Weight::MAX_BLOCK.to_wu() / Weight::MIN_TRANSACTION.to_wu()?

Yeah that could have been better, I went for the Div impl in the latest version